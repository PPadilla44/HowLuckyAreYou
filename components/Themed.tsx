import { Text as DefaultText, View as DefaultView, ScrollViewProps as dScrollViewProps } from 'react-native';
import { Icon as DefaultIcon, IconProps as DefaultIconProps, Input as DefaultInput, InputProps as DefaultInputProps, ButtonGroup as DButtonGroup, ButtonGroupProps as DButtonGroupProps } from 'react-native-elements';
import { ScrollView as DefaultScrollView, NativeViewGestureHandlerProps, TouchableOpacity as DefaultTouchableOpacity } from "react-native-gesture-handler";

import Colors from '../constants/Colors';

import useColorScheme from '../hooks/useColorScheme';

export const useThemeColor = (
    props: { light?: string; dark?: string },
    colorName: keyof typeof Colors.light & keyof typeof Colors.dark
) => {
    // const theme = useColorScheme();
    const theme = "dark";

    const colorFromProps = props[theme];

    return { colorFromProps: colorFromProps ? colorFromProps : Colors[theme][colorName], theme }
}

type ThemeProps = {
    lightColor?: string;
    darkColor?: string;
};

export type TextProps = ThemeProps & DefaultText['props'];
export type ViewProps = ThemeProps & DefaultView['props'];
export type TouchableOpacityProps = ThemeProps & DefaultTouchableOpacity['props'];
export type IconProps = ThemeProps & DefaultIconProps;
export type InputProps = ThemeProps & DefaultInputProps;
export type ScrollViewProps = ThemeProps & NativeViewGestureHandlerProps & dScrollViewProps;
export type ButtonGroupProps = ThemeProps & DButtonGroupProps;

export const Text = (props: TextProps) => {
    const { style, lightColor, darkColor, ...otherProps } = props;
    const color = useThemeColor({ light: lightColor, dark: darkColor }, "text").colorFromProps;

    return <DefaultText style={[{ color, fontFamily: "Futura" }, style]} {...otherProps} />
}

export const View = (props: ViewProps) => {
    const { style, lightColor, darkColor, ...otherProps } = props;
    const backgroundColor = useThemeColor({ light: lightColor, dark: darkColor }, "background").colorFromProps;
    const shadowColor = useThemeColor({  }, "text").colorFromProps;
    return <DefaultView style={[{ backgroundColor, shadowColor }, style]}  {...otherProps} />
}

export const TouchableOpacity = (props: TouchableOpacityProps) => {
    const { style, containerStyle, lightColor, darkColor, ...otherProps } = props;
    const backgroundColor = useThemeColor({ light: lightColor, dark: darkColor }, "button").colorFromProps;
    const oppositeColor = useThemeColor({ }, "text").colorFromProps;
    return <DefaultTouchableOpacity style={[{ backgroundColor, borderColor: oppositeColor, shadowColor: oppositeColor }, style]} containerStyle={[{ backgroundColor, borderColor: oppositeColor }, containerStyle]} {...otherProps} />
}

export const Icon = (props: IconProps) => {
    const { style, lightColor, darkColor, ...otherProps } = props;
    const color = useThemeColor({ light: lightColor, dark: darkColor }, "icon").colorFromProps;
    return <DefaultIcon tvParallaxProperties={undefined} color={color} {...otherProps} />
}

export const Input = (props: InputProps) => {
    const { style, lightColor, darkColor, ...otherProps } = props;
    const { colorFromProps, theme } = useThemeColor({ light: lightColor, dark: darkColor }, "input");
    const backgroundColor = colorFromProps;
    const color = useThemeColor({ light: lightColor, dark: darkColor }, "text").colorFromProps;
    return (

        <>
            <DefaultInput
                style={[{ backgroundColor: "transparent", color }, style]}
                keyboardAppearance={theme}
                {...otherProps}
            />
            <View style={{ backgroundColor, width: "100%", height: "100%", opacity: .5, position: "absolute", borderRadius: 15, zIndex: -10 }} />
        </>
    )
    // return <DefaultInput style={[{backgroundColor: "transparent", color }, style]} keyboardAppearance={theme} {...otherProps} />
}

export const ScrollView = (props: ScrollViewProps) => {
    const { style, darkColor, lightColor, ...otherProps } = props;
    const backgroundColor = useThemeColor({ light: lightColor, dark: darkColor }, "background").colorFromProps;
    return <DefaultScrollView style={[{ backgroundColor }, style]} {...otherProps} />
}

export const ButtonGroup = (props: ButtonGroupProps) => {
    const { containerStyle, darkColor, lightColor, ...otherProps } = props;
    const color = useThemeColor({ light: lightColor, dark: darkColor }, "input").colorFromProps;
    
    return <DButtonGroup containerStyle={[{ backgroundColor: color, borderColor: color }, containerStyle]}  {...otherProps} />
}