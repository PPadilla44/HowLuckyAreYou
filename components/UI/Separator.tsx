import { StyleSheet } from 'react-native'
import React from 'react'
import { View } from '../Themed';
import Colors from '../../constants/Colors';

const Separator = () => <View style={styles.itemSeparator} />;

export default Separator

const styles = StyleSheet.create({
    itemSeparator: {
        flex: 1,
        height: 1,
        backgroundColor: Colors.light.input,
        opacity: .5
    }
})