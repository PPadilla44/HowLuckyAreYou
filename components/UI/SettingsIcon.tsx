import { StyleSheet } from 'react-native'
import React, { FC } from 'react'
import { Icon, TouchableOpacity } from '../Themed'
import Colors from '../../constants/Colors'

interface Props {
    callBack: () => any;
}

const SettingsIcon: FC<Props> = ({ callBack }) => {
    return (
        <TouchableOpacity
            testID='settingsBtn'
            lightColor='white'
            darkColor={Colors.dark.modal}
            onPress={callBack}
            style={styles.btn}
            containerStyle={styles.container} >
            <Icon
                testID='settingsIcon'
                name='tune'
                size={40}
                lightColor={Colors.dark.modal}
                darkColor='white'
            />
        </TouchableOpacity>
    )
}

export default SettingsIcon

const styles = StyleSheet.create({
    btn: {
        borderBottomWidth: 1,
        borderTopWidth: 1,
        borderLeftWidth: 1,
        borderRightWidth: 1,
        borderRadius: 15,
    },
    container: {
        width: 50,
        height: 50,
        position: "absolute",
        right: 10,
        bottom: "5%",
        alignItems: "center",
        justifyContent: "center"
    }
})