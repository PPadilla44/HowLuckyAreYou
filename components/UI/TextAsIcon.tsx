import { StyleSheet } from 'react-native'
import React from 'react'
import { Text } from '../Themed'


const TextAsIcon = ({text}: {text: string}) => {
    return (
        <Text style={styles.icon} >
            {text}
        </Text>
    )
}

export default TextAsIcon

const styles = StyleSheet.create({
    icon: {
        fontSize: 48,
        fontWeight: "bold",
    }
})