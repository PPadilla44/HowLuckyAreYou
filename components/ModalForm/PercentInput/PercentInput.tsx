import { StyleSheet } from 'react-native'
import React, { FC } from 'react'
import { Input, View } from '../../Themed'
import { TextAsIcon } from '../../UI'


export interface Props {
    oddsString: string;
    changeText: (data: {}) => void
}
const PercentInput: FC<Props> = ({ changeText, oddsString }) => {

    return (
        <>
            <View style={styles.percentWrapper}>
                <Input
                    style={styles.input}
                    inputContainerStyle={styles.inputContainer}
                    keyboardType="numeric"
                    placeholder="1.0"
                    maxLength={10}
                    value={oddsString}
                    onChangeText={oddsString => changeText({ oddsString })}
                />
            </View>
            <View style={styles.percentIconContainer}>
                <TextAsIcon style={{fontSize: 48, fontWeight: "bold"}} text={'%'} />
            </View>
        </>
    )
}

export default PercentInput

const styles = StyleSheet.create({
    percentWrapper: {
        flex: 1,
        backgroundColor: "transparent"
    },
    input: {
        borderRadius: 15,
        fontSize: 48,
        fontWeight: "bold",
        fontFamily: "Futura",
        textAlign: "center",
        height: 52,
    },
    percentIconContainer: {
        height: 52,
        marginLeft: 10,
        backgroundColor: "transparent"
    },
    inputContainer: {
        borderBottomWidth: 0,
    },
})