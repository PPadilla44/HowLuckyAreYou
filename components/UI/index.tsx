import Separator from "./Separator";
import TextAsIcon from "./TextAsIcon";
import SettingsIcon from "./SettingsIcon";

export {
    Separator,
    TextAsIcon,
    SettingsIcon
}