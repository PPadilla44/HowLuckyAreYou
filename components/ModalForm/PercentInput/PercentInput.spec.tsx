import { cleanup, fireEvent, render, RenderAPI } from "@testing-library/react-native";
import { TouchableOpacity, View } from "../../Themed";
import { TextAsIcon } from "../../UI";
import PercentInput, { Props } from "./PercentInput";

describe('<PercentInput />', () => {
    
    const mockFn = jest.fn();

    const props : Props = {
        oddsString: "25",
        changeText: mockFn
    }

    let wrapper: RenderAPI;

    afterEach(cleanup)

    beforeEach(() => {
        wrapper = render(<PercentInput {...props} /> )
    });

    it('should render correctly', () => {
        expect(wrapper.container.children).toHaveLength(2);
        expect(wrapper.container.findAllByType(View)).toHaveLength(3);
        expect(wrapper.container.findAllByType(TextAsIcon)).toHaveLength(1);
    });

    it('should render correct placeholder and values', () => {
        wrapper.getByPlaceholderText("1.0");
        wrapper.getByDisplayValue("25");
        wrapper.getByText("%");
    });

    it('should call handleChanges on textChange', () => {
        const input = wrapper.getByPlaceholderText("1.0");
        fireEvent.changeText(input, "testing1");
        expect(mockFn).toBeCalledTimes(1);
    });

    it('should match snapshot', () => {
        expect(wrapper.toJSON()).toMatchSnapshot();
    });

})