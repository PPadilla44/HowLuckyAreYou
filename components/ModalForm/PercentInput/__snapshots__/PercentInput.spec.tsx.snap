// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PercentInput /> should match snapshot 1`] = `
Array [
  <View
    style={
      Array [
        Object {
          "backgroundColor": "black",
          "shadowColor": "white",
        },
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "paddingHorizontal": 10,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderBottomWidth": 0,
            "borderColor": "#86939e",
            "flexDirection": "row",
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          editable={true}
          keyboardAppearance="dark"
          keyboardType="numeric"
          maxLength={10}
          onChangeText={[Function]}
          placeholder="1.0"
          placeholderTextColor="#86939e"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "transparent",
              "borderRadius": 15,
              "color": "white",
              "flex": 1,
              "fontFamily": "Futura",
              "fontSize": 48,
              "fontWeight": "bold",
              "height": 52,
              "minHeight": 40,
              "textAlign": "center",
            }
          }
          testID="percentInput"
          underlineColorAndroid="transparent"
          value="25"
        />
      </View>
      <Text
        style={
          Object {
            "color": "#ff190c",
            "fontSize": 12,
            "margin": 5,
          }
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "black",
            "shadowColor": "white",
          },
          Object {
            "backgroundColor": "#808080",
            "borderRadius": 15,
            "height": "100%",
            "opacity": 0.5,
            "position": "absolute",
            "width": "100%",
            "zIndex": -10,
          },
        ]
      }
    />
  </View>,
  <View
    style={
      Array [
        Object {
          "backgroundColor": "black",
          "shadowColor": "white",
        },
        Object {
          "backgroundColor": "transparent",
          "height": 52,
          "marginLeft": 10,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "white",
            "fontFamily": "Futura",
          },
          Object {
            "fontSize": 48,
            "fontWeight": "bold",
          },
        ]
      }
    >
      %
    </Text>
  </View>,
]
`;
